pkg/platforms/openstack/openstack.go

```diff
@@ -103,7 +104,11 @@ type OSPDeviceInfo struct {
 }
 
 func New() OpenstackInterface {
```

Comment:
New should accept hostManager interface IMO.

similarly to what we do with interfaces and their impl in host pkg

Reply:
+1 it will also help us if we need to pass a mock for unit-tests

you will only need to add the hostManager creation here https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/5887e7522ca8e1ebb13587bce6dafe43d000a992/main.go#L141

the other two places where we call the platform NewDefaultPlatformHelper() already have hostHelper created

Reply:
In service.go and start.go we create the HostHelper, not the HostManager. 

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/5887e7522ca8e1ebb13587bce6dafe43d000a992/cmd/sriov-network-config-daemon/service.go#L105

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/5887e7522ca8e1ebb13587bce6dafe43d000a992/cmd/sriov-network-config-daemon/start.go#L181

I'm not 100% getting the difference between these actors, but I'd leave the helpers's `New` function without arguments, and all the other interfaces get created by injecting the dependencies. WDYT?

Reply:
> and all the other interfaces get created by injecting the dependencies. WDYT?

you will not be able to do so if the test is in a different package. e.g `platform_test`.
that said, this is not the case ATM. so im fine with either.
we can always change later.